objects/src/accounts/mod.rs

```diff
@@ -110,16 +112,12 @@ impl Account {
     /// proofs.
     ///
     /// For existing accounts, this is exactly the same as [Account::hash()], however, for new
-    /// accounts this value is set to [ZERO; 4]. This is because when a transaction is executed
-    /// agains a new account, public input for the initial account state is set to [ZERO; 4] to
+    /// accounts this value is set to `None`. This is because when a transaction is executed
+    /// against a new account, public input for the initial account state is set to `None` to
     /// distinguish new accounts from existing accounts. The actual hash of the initial account
     /// state (and the initial state itself), are provided to the VM via the advice provider.
```

Comment:
The new comment is technically incorrect here - we do still provide `[ZERO; 4]` as a public input into the transaction. Maybe rewrite it as:
```Rust
/// For existing accounts, this is exactly the same as [Account::hash()], however, for new
/// accounts this value is set to `None`. This is because when a transaction is executed
/// against a new account, public input for the initial account state is set to [ZERO; 4] to
/// distinguish new accounts from existing accounts, and `None` returned from this method.
/// indicates that `[ZERO; 4]` should be used as a public input. The actual hash of the initial
/// account state (and the initial state itself), are provided to the VM via the advice provider.
```

But I also don't love this method being here (it is kind of confusing). Since it is used only in the `miden-lib/src/transaction/inputs.rs` file - maybe we move it there and make it into a helper function?